### PR TITLE
Improve changelog targets

### PR DESCRIPTION
**Pull request by wilzbach**
- add different changelog target (pending against `upstream/master` vs. upcoming
release build against `upstream/stable`)
- generate the changelog/next-version, s.t. the file existence check can
be dropped
- move LOOSE_CHANGELOG_FILES to the individual targets
- document the changelog targets
- rename CHANGELOG_VERSION_LATEST to CHANGELOG_VERSION_STABLE
- fix whitespace indentation in the changelog documentation

@MartinNowak currently runs the `changed.d` without this Makefile:

https://gist.github.com/MartinNowak/a471fe7ddbfeef205cdf04c93a94c6d0#file-release-sh-L62

This won't allow us to include the contributor list, hence a separate target as for the release build a different upstream branch is used.

### CODE REVIEW

File: posix.mak [hunk 1]
```diff
@@ -864,23 +864,49 @@ test: $(ASSERT_WRITELN_BIN)_test test/next_version.sh all
...
+#  			make -f posix.mak pending_changelog
+#
+#  	  This will look at changes up to `upstream/master` and can be used to
+#  	  preview the changelog locally.
+#
+#  b) Generate the changelog for an upcoming release
+#
+#  		make -f posix.mak prerelease_changelog
+#
+#   This will look at changes to upstream/stable and is run by the release manager.
```
Comment (JinShil) [inline posix.mak]: Seems like a lot of gratuitous whitespace in those comments.

Comment (wilzbach) [reply posix.mak]: Yes, this change still used tabs. Good catch!

File: posix.mak [hunk 2]
```diff
@@ -849,38 +849,67 @@ test: $(ASSERT_WRITELN_BIN)_test test/next_version.sh all
...
+$G/changelog/next-version: ${DMD_DIR}/VERSION
 	$(eval NEXT_VERSION:=$(shell changelog/next_version.sh ${DMD_DIR}/VERSION))
+	@mkdir -p $(dir $@)
+	@echo $(NEXT_VERSION) > $@
 
-changelog/pending.dd: changelog/next-version | ${STABLE_DMD} ../tools ../installer
-	[ -f changelog/pending.dd ] || $(STABLE_RDMD) -version=Contributors_Lib $(TOOLS_DIR)/changed.d \
-		$(CHANGELOG_VERSION_LATEST) -o changelog/pending.dd --version "${NEXT_VERSION}" \
+changelog/prerelease.dd: $G/changelog/next-version $(LOOSE_CHANGELOG_FILES) | ${STABLE_DMD} ../tools ../installer
+	$(STABLE_RDMD) version=Contributors_Lib $(TOOLS_DIR)/changed.d \
```
Comment (CyberShadow) [inline posix.mak]:
```
Error: module version=Contributors_Lib is in file 'version=Contributors_Lib.d' which cannot be read
```

Looks like a `-` went missing.

**Comment by wilzbach** [comment]
> Looks like a - went missing.

Yes. Thanks. Fixed!

File: posix.mak [hunk 3]
```diff
@@ -858,40 +858,69 @@ test: $(ASSERT_WRITELN_BIN)_test test_dspec test/next_version.sh all
...
-		$(CHANGELOG_VERSION_LATEST) -o changelog/pending.dd --version "${NEXT_VERSION}" \
+changelog/prerelease.dd: $G/changelog/next-version $(LOOSE_CHANGELOG_FILES) | ${STABLE_DMD} ../tools ../installer
+	$(STABLE_RDMD) -version=Contributors_Lib $(TOOLS_DIR)/changed.d \
+		$(CHANGELOG_VERSION_STABLE) -o $@ --version "${NEXT_VERSION}" \
 		--date "To be released"
 
-pending_changelog: $(LOOSE_CHANGELOG_FILES) changelog/pending.dd html
+changelog/pending.dd: $G/changelog/next-version $(LOOSE_CHANGELOG_FILES) | ${STABLE_DMD} ../tools ../installer
+	$(STABLE_RDMD) -version=Contributors_Lib $(TOOLS_DIR)/changed.d \
+		$(CHANGELOG_VERSION_MASTER) -o $@ --version "${NEXT_VERSION}" \
```
Comment (wilzbach) [inline posix.mak]: This change is the most important one as currently the pending changelog is built against `upstream/stable`:

https://dlang.org/changelog/pending.html

**Pull request merged**